src/EffectTransform.js

```diff
@@ -217,12 +217,15 @@ class EffectTransform {
         dst[1] = (dst[1] - skinUniforms.u_logicalBounds[1]) /
             (skinUniforms.u_logicalBounds[3] - skinUniforms.u_logicalBounds[1]);
 
-        if ((effects & ShaderManager.EFFECT_INFO.mosaic.mask) !== 0) {
+        const pointInsideLogicalBounds = dst[0] >= 0 && dst[0] <= 1 && dst[1] >= 0 && dst[1] <= 1;
```

Comment:
Would it make sense to have the caller do the remap to logical coordinates and then this check, and not even call `transformPoint` if the logical coordinate is outside the bounds? I'm worried about side effects of discontinuous transformations in (for example) convex hull calculation.

Reply:
Done in https://github.com/LLK/scratch-render/pull/594/commits/036895581e72045b95957e656eec64f117e8eca5